backend/routers/batches.py

```
@@ -0,0 +1,596 @@
+        await db.batch_items.delete_many({"batch_id": batch_id})
+        await db.batches.delete_one({"id": batch_id})
+        if idem_hash:
+            existing = await db.batches.find_one(
+                {"user_id": user.id, "idempotency_hash": idem_hash}
+            )
+            if existing:
+                if existing.get("request_fingerprint") != request_fingerprint:
+                    raise HTTPException(
+                        status_code=409,
+                        detail="Idempotency-Key was already used for a different batch request",
+                    )
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)